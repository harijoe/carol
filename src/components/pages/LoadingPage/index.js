import React from 'react'
import PropTypes from 'prop-types'
import styled, { ThemeProvider } from 'styled-components'
import injectTranslate from 'i18n/hoc/injectTranslate'

import { Icon, Heading } from 'components'
import { QuotatisLogoStatic } from 'containers'
import defaultTheme from '../../themes/default'

const LoadingWrapper = styled.div`
  display: flex;
  height: 100%;
  padding: 0;
  margin: 0;
  align-items: center;
  justify-content: center;
  min-height: 100vh;
`

const Row = styled.div`
  width: auto;
`

const Element = styled.div`
  text-align: center;
  margin-top: 30px;
`

const Wrapper = styled.div`
  min-height: 100vh;
`

const LoadingPage = ({ translate }) => (
  <ThemeProvider theme={defaultTheme}>
    <Wrapper>
      <QuotatisLogoStatic />
      <LoadingWrapper>
        <Row>
          <Element><Icon icon="spinner" size={100} /></Element>
          <Element><Heading level={2}>{translate('loading.message')}</Heading></Element>
        </Row>
      </LoadingWrapper>
    </Wrapper>
  </ThemeProvider>
)

LoadingPage.propTypes = {
  translate: PropTypes.func,
}

export default injectTranslate(LoadingPage)
