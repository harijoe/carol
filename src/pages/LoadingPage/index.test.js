import React from 'react'
import { shallow } from 'enzyme'
import LoadingPage from './'

it('renders LoadingPage', () => {
  expect(shallow(<LoadingPage />)).toMatchSnapshot()
})
