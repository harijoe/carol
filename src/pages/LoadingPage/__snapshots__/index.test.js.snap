// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders LoadingPage 1`] = `
<LoadingPage
  labelWithColon={[Function]}
  rawTranslate={[Function]}
  translate={[Function]}
/>
`;
